### PR TITLE
fixes extended test issues caused by #11411

### PR DESCRIPTION
**Pull request by coreydaley**

### CODE REVIEW

**Comment by coreydaley** [comment]
[testextended]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

File: test/extended/builds/forcepull.go [hunk 1]
```diff
@@ -38,6 +38,16 @@ func doTest(bldPrefix, debugStr string, same bool, oc *exutil.CLI) {
 	// corrupt the builder image
 	exutil.CorruptImage(fullImageName, corruptor)
 
+	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+		// expect custom builds to fail by default
+		_, err := exutil.StartBuildAndWait(oc, bldPrefix)
+		o.ExpectWithOffset(1, err).To(o.HaveOccurred())
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: we don't need to confirm that custom builds are disabled by default here, i assume/hope we have other tests for that somewhere as part of your PR to disable them?

so just add the role.

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
@bparees made your suggested change, and tests are passing, please review

File: test/extended/builds/forcepull.go [hunk 2]
```diff
@@ -53,6 +59,11 @@ func doTest(bldPrefix, debugStr string, same bool, oc *exutil.CLI) {
 	// dump tags/hexids for debug
 	_, err = exutil.DumpAndReturnTagging(tags)
 	o.ExpectWithOffset(1, err).NotTo(o.HaveOccurred())
+
+	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+		err = oc.AsAdmin().Run("adm").Args("policy", "remove-cluster-role-from-user", "system:build-strategy-custom", oc.Username()).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+	}
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: I think this should be setup in a defer block so we run it even if the test fails.  same for the one below.

**Comment by coreydaley** [comment]
@bparees updated per your suggestion and tests are passing.  Please review.

File: test/extended/builds/forcepull.go [hunk 3]
```diff
@@ -53,6 +59,13 @@ func doTest(bldPrefix, debugStr string, same bool, oc *exutil.CLI) {
 	// dump tags/hexids for debug
 	_, err = exutil.DumpAndReturnTagging(tags)
 	o.ExpectWithOffset(1, err).NotTo(o.HaveOccurred())
+
+	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+		defer func() {
+			err = oc.AsAdmin().Run("adm").Args("policy", "remove-cluster-role-from-user", "system:build-strategy-custom", oc.Username()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}()
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: you need to setup the defer immediately after adding the role, the defer has to be in place before you start running the actual test steps.

File: test/extended/builds/forcepull.go [hunk 4]
```diff
@@ -38,6 +38,12 @@ func doTest(bldPrefix, debugStr string, same bool, oc *exutil.CLI) {
 	// corrupt the builder image
 	exutil.CorruptImage(fullImageName, corruptor)
 
+	if bldPrefix == buildPrefixFC || bldPrefix == buildPrefixTC {
+		// grant access to the custom build strategy
+		err := oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:build-strategy-custom", oc.Username()).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
```
Comment (bparees) [inline test/extended/builds/forcepull.go]: that is, put the defer right here.

**Comment by coreydaley** [comment]
@bparees updated, please review.

**Comment by bparees** [comment]
lgtm, thanks @coreydaley!
[merge]

**Pull request merged**